src/mozanalysis/experiment.py

```diff
@@ -563,6 +626,74 @@ def _build_enrollments_query_glean_event(self, time_limits, dataset):
             dataset=self.app_id or dataset,
         )
 
+    def _build_exposure_query_normandy(self, time_limits):
+        """Return SQL to query exposures for a normandy experiment"""
+        return """
+        SELECT
+            e.client_id,
+            e.branch,
+            min(e.submission_date) AS exposure_date,
+            COUNT(e.submission_date) AS num_exposure_events
+        FROM raw_enrollments re
+        LEFT JOIN (
+            SELECT
+                client_id,
+                `mozfun.map.get_key`(event_map_values, 'branch') AS branch,
+                submission_date
+            FROM
+                `moz-fx-data-shared-prod.telemetry.events`
+            WHERE
+                event_category = 'normandy'
+                AND (event_method = 'exposure' AND event_method = 'expose')
+                AND submission_date
+                    BETWEEN '{first_enrollment_date}' AND '{last_enrollment_date}'
+                AND event_string_value = '{experiment_slug}'
+        ) e
+        ON re.client_id = e.client_id AND
+            re.branch = e.branch AND
+            e.submission_date >= re.enrollment_date
+        GROUP BY e.client_id, e.branch
+            """.format(
+            experiment_slug=self.experiment_slug,
+            first_enrollment_date=time_limits.first_enrollment_date,
+            last_enrollment_date=time_limits.last_enrollment_date,
+        )
+
+    def _build_exposure_query_glean_event(self, time_limits, dataset):
+        """Return SQL to query exposures for a Glean no-event experiment"""
+        return """
+            SELECT
+                e.client_id,
+                e.branch,
+                DATE(MIN(e.submission_date)) AS exposure_date,
+                COUNT(e.submission_date) AS num_exposure_events
+            FROM raw_enrollments re
+            LEFT JOIN (
+                SELECT
+                    client_info.client_id,
+                    `mozfun.map.get_key`(e.extra, 'branch') AS branch,
+                    DATE(events.submission_timestamp) AS submission_date
+                FROM
+                    `moz-fx-data-shared-prod.{dataset}.events` events,
+                    UNNEST(events.events) AS e
+                WHERE
+                    DATE(events.submission_timestamp)
+                    BETWEEN '{first_enrollment_date}' AND '{last_enrollment_date}'
+                    AND e.category = "nimbus_events"
+                    AND mozfun.map.get_key(e.extra, "experiment") = '{experiment_slug}'
+                    AND (e.name = 'expose' OR e.name = 'exposure')
+            ) e
```

Comment:
nit: a couple of different things are aliased to `e` here; can they take different names?